pkg/info/info.go

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) The Thanos Authors.
+// Licensed under the Apache License 2.0.
+
+package info
+
+import (
+	"context"
+
+	"github.com/go-kit/kit/log"
+	"github.com/thanos-io/thanos/pkg/info/infopb"
+	"github.com/thanos-io/thanos/pkg/store/labelpb"
+	"google.golang.org/grpc"
+)
+
+type InfoServer struct {
+	infopb.UnimplementedInfoServer
+
+	logger    log.Logger
+	component infopb.ComponentType
+
+	getLabelSet        func() []labelpb.ZLabelSet
+	getStoreInfo       func() *infopb.StoreInfo
+	getExemplarsInfo   func() *infopb.ExemplarsInfo
+	ruleAPIPresent     bool
+	targetAPIPresent   bool
+	metadataAPIPresent bool
+}
+
+func NewInfoServer(
+	component infopb.ComponentType,
+	getLabelSet func() []labelpb.ZLabelSet,
+	getStoreInfo func() *infopb.StoreInfo,
+	getExemplarsInfo func() *infopb.ExemplarsInfo,
+	ruleAPIPresent bool,
```

Comment:
Let's make all of these parameters follow the same pattern, so that rather than passing a func for some and a bool for the other, they all accept a func. This will present a more coherent API for users and also ensure that we don't need to change this signature when we add fields, e.g. to ruleAPIInfo, in the future.